File: app/models/questions/SimpleQuestion.java
package models.questions;

import play.Logger;

public class SimpleQuestion extends Question {

	private String answer;
	
	public SimpleQuestion(int questionNumber, String question, String answer) {
		super(AnswerType.SIMPLE, questionNumber, question);
		this.answer = answer;
	}

	@Override
	public boolean checkAnswer(String answer) {
		Logger.debug("checking answer " + answer + " against " + this.answer);
		return answer.equalsIgnoreCase(this.answer);
	}

}

File: appKotlin/models/questions/SimpleQuestion.kt
package models.questions

class SimpleQuestion(questionNumber: Int, question: String, private val answer: String) : models.questions.Question(models.questions.Question.AnswerType.SIMPLE, questionNumber, question) {

    override fun checkAnswer(answer: String): Boolean {
        play.Logger.debug("checking answer " + answer + " against " + this.answer)
        return answer.equals(this.answer, ignoreCase = true)
    }

}
